### PR TITLE
Change all uses of MappingType and mapping_type

### PR DESCRIPTION
**Pull request by GrahamBenHarper**
I've heard the use of MappingType in mapping.h is a little confusing since it's not actually a type.

To fix that, I've changed all instances of MappingType to MappingKind, all instances of mapping_type to mapping_kind, and then updated as much documentation as I could find to accordingly match. Please let me know if I missed anything.

I'm running a build on my end now to verify nothing was broken but it may take awhile on my laptop :cry:

### CODE REVIEW

**Comment by bangerth** [comment]
Do you think you could add an entry to `doc/news/changes/minor` that explains your change?

**Comment by GrahamBenHarper** [comment]
I have added a changelog entry.

Also, I'd like feedback: the file `bundled/boost-1.70.0/include/boost/geometry/strategies/geographic/mapping_ssf.hpp` has its own enum named `mapping_type`. I did not modify it because it doesn't use camel case, which is used for types, but I can change it also if you'd like.

**Comment by tjhei** [comment]
> I did not modify it because it doesn't use camel case, which is used for types, but I can change it also if you'd like.

This should stay as is.  We do not develop/improve things in ``bundled/`` (these are basically fixed, imported versions).

**Comment by masterleinad** [comment]
Is this backward-compatible?

**Comment by GrahamBenHarper** [comment]
Backward compatible in which sense?

I only renamed the enum and all references to the enum, so I imagine it should be.

**Comment by bangerth** [comment]
It's not backward compatible: The `enum MappingType/Kind` is a public one in the mapping and finite element classes. At the same time, it's only used internally in these kinds of classes, so I suspect that the likelihood of anyone using the type in user classes is, essentially, zero.

@GrahamBenHarper -- can you move the changelog entry to the `doc/news/changes/incompatible` directory?

**Comment by GrahamBenHarper** [comment]
Oh I understand what you mean now! I never would have thought of that.

The changelog entry should be in the right place now :)

File: doc/news/changes/incompatibilities/20190806GrahamHarper [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Changed: Mapping.h referred to different kinds of mappings in an enum
```
Comment (masterleinad) [inline doc/news/changes/incompatibilities/20190806GrahamHarper]:
```suggestion
Changed: mapping.h referred to different kinds of mappings in an enum 
```

File: doc/news/changes/minor/20190806GrahamHarper [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Changed: Mapping.h referred to different kinds of mappings in an enum MappingType; however, since it's an enum, MappingType is a misleading name. All instances of MappingType and mapping_type have been changed to MappingKind and mapping_kind, respectively. The documentation has been updated in order to reflect these changes as well.
+<br>
+(Graham Harper, 2019/08/06)
```
Comment (masterleinad) [inline doc/news/changes/minor/20190806GrahamHarper]: Just delete this file.

**Comment by GrahamBenHarper** [comment]
Whoops!

I also took the liberty of squashing those multiple egregious commits. The changelog entry should be in the right place now.

File: doc/news/changes/incompatibilities/20190806GrahamHarper (hunk 1 above)
Comment (GrahamBenHarper) [reply doc/news/changes/incompatibilities/20190806GrahamHarper]: Got the typo!

File: doc/news/changes/minor/20190806GrahamHarper (hunk 2 above)
Comment (GrahamBenHarper) [reply doc/news/changes/minor/20190806GrahamHarper]: Deleted!

**Pull request merged**